pennylane/_grad.py

```diff
@@ -21,14 +21,19 @@
 from autograd.numpy.numpy_boxes import ArrayBox
 from autograd.extend import vspace
 from autograd.wrap_util import unary_to_nary
+from pennylane.compiler import compiler
 
 make_vjp = unary_to_nary(_make_vjp)
 
 
 class grad:
-    """Returns the gradient as a callable function of (functions of) QNodes.
+    """A :func:`~.qjit` compatible gradient transformation that returns the gradient
+    as a callable function of (functions of) QNodes.
 
-    By default, gradients are computed for arguments which contain the property
+    Inside a :func:`~.qjit` decorated program, this will patch to :func:`catalyst:.grad` and
```

Comment:
@josh146 Do you have any idea why references to the catalyst doc don't parse properly? Here's the link to the `doc/conf.py` https://github.com/PennyLaneAI/pennylane/blob/657dd62c72945e3903ef1fbea474f8c3fe064193/doc/conf.py#L67

It doesn't show and link to `catalyst` doc pages, 
![image](https://github.com/PennyLaneAI/pennylane/assets/10773383/4ed0a431-c6ce-4d14-a9f6-3d9f8d6f70f3)

Reply:
@maliasadi maybe 
```
:func:`catalyst.grad` or :func:`catalyst:catalyst.grad`
```
would work?

Reply:
Thanks @josh146! It only worked with, 
```
:func:`catalyst.grad`
```